quantum/rgb_matrix_animations/typing_heatmap_anim.h

```diff
@@ -2,72 +2,68 @@
 RGB_MATRIX_EFFECT(TYPING_HEATMAP)
 #ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS
 
-void process_rgb_matrix_typing_heatmap(keyrecord_t *record) {
-    uint8_t row = record->event.key.row;
-    uint8_t col = record->event.key.col;
-    uint8_t m_row = row - 1;
-    uint8_t p_row = row + 1;
-    uint8_t m_col = col - 1;
-    uint8_t p_col = col + 1;
+// Array not large enough for PROGMEM to save firmware space on AVR
+const uint8_t k_heatmap_vals[9] = {
+    13, 16, 13,
+    16, 32, 16,
+    13, 16, 13
+};
 
-    if (m_col < col)
-      rgb_frame_buffer[row][m_col] = qadd8(rgb_frame_buffer[row][m_col], 16);
-    rgb_frame_buffer[row][col] = qadd8(rgb_frame_buffer[row][col], 32);
-    if (p_col < MATRIX_COLS)
-      rgb_frame_buffer[row][p_col] = qadd8(rgb_frame_buffer[row][p_col], 16);
+// TODO: Create a variable blackboard / scratchpad (memory page) that effects can use
+static uint32_t heatmap_timer = 0;
 
-    if (p_row < MATRIX_ROWS) {
-      if (m_col < col)
-        rgb_frame_buffer[p_row][m_col] = qadd8(rgb_frame_buffer[p_row][m_col], 13);
-      rgb_frame_buffer[p_row][col] = qadd8(rgb_frame_buffer[p_row][col], 16);
-      if (p_col < MATRIX_COLS)
-        rgb_frame_buffer[p_row][p_col] = qadd8(rgb_frame_buffer[p_row][p_col], 13);
-    }
+static inline void heatmap_reset_timer(void) {
+    uint8_t invert_spd = UINT8_MAX - rgb_matrix_config.speed;
+    heatmap_timer = g_rgb_timer + 16 + invert_spd / 4;
+}
 
-    if (m_row < row) {
-      if (m_col < col)
-        rgb_frame_buffer[m_row][m_col] = qadd8(rgb_frame_buffer[m_row][m_col], 13);
-      rgb_frame_buffer[m_row][col] = qadd8(rgb_frame_buffer[m_row][col], 16);
-      if (p_col < MATRIX_COLS)
-        rgb_frame_buffer[m_row][p_col] = qadd8(rgb_frame_buffer[m_row][p_col], 13);
+void process_rgb_matrix_typing_heatmap(keyrecord_t *record) {
```

Comment:
Could it be possible to add the ability to disable the heatmap effect on key-release, as well?  

Should be simple as: 
```suggestion
void process_rgb_matrix_typing_heatmap(keyrecord_t *record) {

#ifndef RGB_MATRIX_KEYRELEASES
    if (!record->event.pressed) { return; }
#endif
```

Reply:
good point, I'll update the code to make sure it plays nice with pressed vs released

Reply:
Fixed in latest commit